src/timer.tsx

```diff
@@ -0,0 +1,438 @@
+        minutes: texts.timerMinutesShortLabel,
+        seconds: texts.timerSecondsShortLabel,
+    };
+
+    const unitLabel: {[key in TimeUnit]: string} = {
+        days: texts.timerDayLongLabel,
+        hours: texts.timerHourLongLabel,
+        minutes: texts.timerMinuteLongLabel,
+        seconds: texts.timerSecondLongLabel,
+    };
+
+    const unitLabelPlural: {[key in TimeUnit]: string} = {
+        days: texts.timerDaysLongLabel,
+        hours: texts.timerHoursLongLabel,
+        minutes: texts.timerMinutesLongLabel,
+        seconds: texts.timerSecondsLongLabel,
+    };
+
+    const renderFormattedNumber = (value: number) => {
+        const digitCount = Math.max(String(value).length, labelType === 'long' ? 1 : 2);
+
+        // Set container's minWidth in ch to avoid it from updating it's width when numbers change
+        return (
+            <span className={styles.unitContainer} style={{minWidth: `${digitCount}ch`}}>
+                {String(value).padStart(digitCount, '0')}
+            </span>
+        );
+    };
+
+    const renderTime = () => {
+        switch (labelType) {
+            case 'none':
+                return timerValue.map((item, index) => (
+                    <React.Fragment key={index}>
+                        {index > 0 && ':'}
+                        {renderFormattedNumber(item.value)}
+                    </React.Fragment>
+                ));
+
+            case 'short':
+                return timerValue.map((item, index) => (
+                    <React.Fragment key={index}>
+                        {index > 0 && ' '}
+                        <span style={{display: 'inline-block'}}>
+                            {renderFormattedNumber(item.value)}
+                            {` ${unitShortLabel[item.unit]}`}
+                        </span>
+                    </React.Fragment>
+                ));
+
+            case 'long':
+            default:
+                return timerValue.map((item, index) => (
+                    <React.Fragment key={index}>
+                        {index > 0 && ' '}
+                        {renderFormattedNumber(item.value)}
+                        {` ${item.value === 1 ? unitLabel[item.unit] : unitLabelPlural[item.unit]}`}
+                        {index === timerValue.length - 2 && ` ${texts.timerAnd}`}
+                        {index < timerValue.length - 2 && ','}
+                    </React.Fragment>
+                ));
+        }
+    };
+
+    const timerLabel = timerValue
+        .map(
+            (item, index) =>
+                `${item.value} ${item.value === 1 ? unitLabel[item.unit] : unitLabelPlural[item.unit]}${
+                    index === timerValue.length - 1
+                        ? ''
+                        : index === timerValue.length - 2
+                        ? ` ${texts.timerAnd} `
+                        : ', '
+                }`
+        )
+        .join('');
+
+    return (
+        <div
+            role="timer"
```

Comment:
how is this announced by screen reader?
should we allow configuring `aria-live` prop? https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/timer_role#associated_wai-aria_roles_states_and_properties

Reply:
Not needed. When setting the role="timer", the aria-live is automatically set to "off". And this is expected, we don't want it to announce changes every second